comp/core/gui/guiimpl/systray/Sources/WiFiIPCServer.swift

```diff
@@ -0,0 +1,242 @@
+import Foundation
+
+/// Request structure for IPC commands
+struct WiFiIPCRequest: Codable {
+    let command: String
+}
+
+/// Response structure for IPC
+struct WiFiIPCResponse: Codable {
+    let success: Bool
+    let data: WiFiData?
+    let error: String?
+}
+
+/// WiFiIPCServer handles Unix socket communication for WiFi data
+class WiFiIPCServer {
+    private let wifiDataProvider: WiFiDataProvider
+    private var socketFileDescriptor: Int32 = -1
+    private var isRunning = false
+    private var acceptQueue: DispatchQueue
+    private let socketPath: String
+
+    init(wifiDataProvider: WiFiDataProvider) {
+        self.wifiDataProvider = wifiDataProvider
+
+        // Create socket path based on current user's UID
+        let uid = getuid()
+        self.socketPath = "/var/run/datadog-agent/wifi-\(uid).sock"
+        self.acceptQueue = DispatchQueue(label: "com.datadoghq.wifi.ipc", qos: .userInitiated)
```

Comment:
Is `.userInitiated` QoS appropriate? This is background telemetry, not user-facing. Consider `.utility` or `.background`?